modules/json/src/org/apache/axis2/json/gson/GsonXMLStreamReader.java

```diff
@@ -780,19 +780,20 @@ private String nextValue() {
      * @return true if it matches false otherwise
      */
     private boolean validateArgumentTypes(JsonToken tokenType, String expectedType) {
-        if (expectedType.equalsIgnoreCase(tokenType.toString())) {
+        if (expectedType != null && expectedType.equalsIgnoreCase(tokenType.toString())) {
             return true;
         } else if (tokenType == JsonToken.NULL) {
             return true;
         } else if (tokenType == JsonToken.NUMBER) {
-            if (expectedType.equals("int") || expectedType.equals("long") || expectedType.equals("double")
-                    || expectedType.equals("float") || expectedType.equals("decimal") || expectedType.equals("byte")
-                    || expectedType.equals("short")) {
+            if (expectedType != null && (expectedType.equals("int") || expectedType.equals("long")
+                    || expectedType.equals("double") || expectedType.equals("float") || expectedType.equals("decimal")
+                    || expectedType.equals("byte") || expectedType.equals("short"))) {
```

Comment:
shall we do the check like _"int".equals(expectedType)_ . So that we can avoid null check at the beginning.